range.go

```diff
@@ -217,6 +243,144 @@ func splitComparatorVersion(s string) (string, string, error) {
 	return strings.TrimSpace(s[0:i]), s[i:], nil
 }
 
+// getWildcardType will return the type of wildcard that the
+// passed version contains
+func getWildcardType(vStr string) wildcardType {
+	parts := strings.Split(vStr, ".")
```

Comment:
Might not work for prerelease and build metadata version information since it may contain dots like '1.0.0-1.x.x' which is a valid version:
http://semver.org/#spec-item-9